packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/schedule.ts

```diff
@@ -0,0 +1,121 @@
+import { Duration } from '../../core';
+
+/**
+ * Schedule for EventBridge Scheduler
+ */
+export class Schedule {
+
+  /**
+   * Construct a schedule from a Date.
+   */
+  public static oneTime(time: Date): Schedule {
+    const pad = (num: number) => (num < 10 ? '0' + num : num);
+
+    const year = time.getFullYear();
+    const month = pad(time.getMonth() + 1);
+    const day = pad(time.getDate());
+    const hours = pad(time.getHours());
+    const minutes = pad(time.getMinutes());
+    const seconds = pad(time.getSeconds());
+
+    return new Schedule(`${year}-${month}-${day}T${hours}:${minutes}:${seconds}`);
+  }
+
+  /**
+   * Construct a schedule from an interval.
+   *
+   * The minimum interval is 1 minute.
+   */
+  public static rate(duration: Duration): Schedule {
+    if (duration.toMilliseconds() < Duration.minutes(1).toMilliseconds()) {
+      throw new Error('Duration cannot be less than 1 minute');
+    }
+
+    const maybeRate = (value: number, unit: string) => (value > 0 && Number.isInteger(value)) ? `${value} ${unit}` : undefined;
+
+    let rate = maybeRate(duration.toDays({ integral: false }), 'days');
```

Comment:
This implementation is derived from existing [Schedule class](https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-events/lib/schedule.ts#L38)

This part of the implementation was added during the community review.
https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-events/lib/schedule.ts#L38

I think they may be using the word "maybe" to imply that "this rate might be applicable for this unit, but it's not necessarily certain."

Since I'm not very familiar with English, I would appreciate it if you could suggest a suitable name for this.

Reply:
Ohhh LOL it's saying that it "may return a rate" if it's a whole number, otherwise it will return undefined. Ok I think that is fine if there's an existing naming pattern already. Could you just add a comment above line 34 to explain that the `maybeRate` method `Returns the rate if the rate is whole number`